### PR TITLE
[MRG] add open_browser trait to ExtensionApp

### PR DESCRIPTION
**Pull request by Zsailer**
Addresses #120. 

Adds a trait to `ExtensionApp`s to configure opening browser windows on launch.

### CODE REVIEW

**Comment by vidartf** [comment]
Should this also get a flag (inverted): `--no-browser` ?

**Comment by Zsailer** [comment]
The `ExtensionApp` actually already inherits the `--no-browser` flag from ServerApp before this PR (see [here](https://github.com/jupyter/jupyter_server/blob/5476f7cf257a5af18272090115d94e76a82ddd86/jupyter_server/extension/application.py#L14)).

**Comment by vidartf** [comment]
That inherited flag reads as:
```python
flags["no-browser"] = (
    {"ServerApp": {"open_browser": False}},
    _("Prevent the opening of the default url in the browser."),
)
```
Is it not an issue that it says `ServerApp` (the extension app does not inherit it)? If so, I imagine some other flags/aliases might have an issue too.

**Comment by echarles** [comment]
@vidartf As far as I have seen, the flags handling does not work as expected. I am looking at this  and will open on issue (hopefully a PR). We can revisit this PR based on the findings/fixes we will get for the more global flag issue.

**Comment by echarles** [comment]
@Zsailer @vidartf 

This branch fails on `jupyter server --open-browser false` `[C 11:44:23.991 ServerApp] Unrecognized flag: '--open-browser'`

I had that issue at other places (https://github.com/jupyter/jupyter_server/pull/117
and https://github.com/jupyterlab/jupyterlab/pull/7416).

I have opened https://github.com/jupyter/jupyter_server/issues/140 to further discuss this before coming back here.

File: jupyter_server/extension/application.py [hunk 1]
```diff
@@ -140,6 +141,11 @@ def static_url_prefix(self):
         help=_("The default URL to redirect to from `/`")
     )
 
+    open_browser = Bool(
+        True,
+        help=_("Should the extension open a browser window?")
+    )
+
```
Comment (echarles) [inline jupyter_server/extension/application.py]: Do we want an alias on this open_browser config so we can use `jupyter server --open-browser False`?

```
    aliases.update({
        'open-browser': 'ExtensionApp.open_browser',
    })
```

**Comment by vidartf** [comment]
@echarles  That is not how it is intended to be used though. `jupyter notebook --open-browser false` also fails.

**Comment by vidartf** [comment]
PS: The right ways are either `--NotebookApp.open_browser=False` or via the flag `--no-browser`.

**Comment by vidartf** [comment]
PPS: Or in this case `--ServerApp.open_browser=False` or the flag.

**Comment by echarles** [comment]
After yesterday discussion at server weekly meeting, I now get the point that the extension flags should only be applied to an extension launched via its entrypoint.

BTW, this PR does not define flag for open_browser, so my previous attempt was out of context... Sorry for the confusion.

Going back to the parameters approach, I have tried to launch with `--ServerApp.open_browser=False` or `--ExtensionApp.open_browser=False` and it does not work (the browser if opened and it should not be).

Digging a bit more into the base ExtensionApp, I have the impression that the traits given via CLI are not taken into account by the ExtensionApp. e.g. `jupyter simple-ext1 --ServerApp.custom_display_url=foo` has no impact on the displayed URL.

Looking at the code that prepare the settings and printing those settings in the log, I feel that something must be done to feed the extension CLI config to the settings. Any input on this?

File: jupyter_server/extension/application.py (hunk 1 above)
Comment (Zsailer) [reply jupyter_server/extension/application.py]: Hm, I think the `--no-browser` flag is probably enough.

The main `ServerApp` doesn't have an `open-browser` alias, so I think we should keep parity with that.

**Comment by Zsailer** [comment]
> Is it not an issue that it says ServerApp (the extension app does not inherit it)? If so, I imagine some other flags/aliases might have an issue too.

@vidartf I don't think this is a problem. These traits *should* affect the ServerApp directly. 

`open_browser` is a special case because ExtensionApps can affect the `ServerApp.open_browser` trait using it's own `open_browser` trait.

**Comment by Zsailer** [comment]
Merging, since I've addressed the concerns above. We can open a future PR if needed.

**Pull request merged**